### ISSUE TITLE
Add background job processor.

### ISSUE THREAD

**Issue by papodaca** [mayeaux/nodetube#62]
If the intent of this application is to be scaleable, it isn't ideal to process all the uploaded files in the main web server process. After a few (100-1000 not sure) simultaneous uploads performance may suffer handling other requests.

I've found a few options for achieving this easily:
* [kue](https://github.com/Automattic/kue)
* [node-resque](https://github.com/actionhero/node-resque)
* [null](https://github.com/OptimalBits/bull)

To achieve this the code in [this](https://github.com/mayeaux/nodetube/blob/master/controllers/backend/uploading.js#L209) async block would be moved to a background job.

**Comment by mayeaux** [comment]
The current approach I have to this, and let me know your take, is that the app is kind of partitioned into different sections. Check the top of .env.settings.sample, you can see :
```
LIVESTREAM_APP=false
FILE_HOST=false
FRONTEND=true
```
Basically you can run the app just as a file host which receives uploads, and the frontend is decoupled and can run as a standalone app.

Is that what you were getting at?

**Comment by dmdxv** [comment]
This is something we should look at if we plan on scaling nodetube.live

**Comment by papodaca** [comment]
@mayeaux Not sure I follow 100% of what you are saying. Does this FILE_HOST/FRONTEND flag have to do with that http path certain processes are responding to? If that is the case at larger scales this may be insufficient.
This isn't a feature I require or anything just an architecture suggestion that I thought might be helpful.

**Comment by mayeaux** [comment]
@papodaca Yup, you can see it here: https://github.com/mayeaux/nodetube/blob/master/config/routes.js

So you can decouple the frontend from other functionality, I agree eventually a job queue will be needed for scale but it's at least something for now

**Comment by Yemeni** [comment]
we are going to need this too for https://github.com/mayeaux/nodetube/pull/341 to resume quality convert
